### PR TITLE
Support UserManagementService on the Java bindings

### LINKED ISSUE #12401: java bindings: add UserManagementServiceClient

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
[Java bindings] User management service client added
changelog_end

Fixes #12401

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/LedgerServices.scala [hunk 1]
```diff
@@ -234,6 +234,17 @@ final class LedgerServices(val ledgerId: String) {
     }
   }
 
+  def withUserManagementClient(
+      authService: AuthService = AuthServiceWildcard,
+      accessToken: java.util.Optional[String] = java.util.Optional.empty[String],
+  )(f: (UserManagementClientImpl, UserManagementServiceImpl) => Any): Any = {
```
Comment (S11001001) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/LedgerServices.scala:240]: 😢

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/UserManagementServiceImpl.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+final class UserManagementServiceImpl extends UserManagementService with FakeAutoCloseable {
+
+  private val log = mutable.ArrayBuffer.empty[Any]
+
+  def requests(): Array[Any] = {
+    val requests = Array.ofDim[Any](log.size)
+    log.copyToArray(requests)
+    requests
```
Comment (S11001001) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/UserManagementServiceImpl.scala:23]:
```suggestion
    log.toArray
```

```scala
scala> val b = mutable.ArrayBuffer.empty[String]
val b: scala.collection.mutable.ArrayBuffer[String] = ArrayBuffer()

scala> b += "hi"
val res0: b.type = ArrayBuffer(hi)

scala> val hi = b.toArray
val hi: Array[String] = Array(hi)

scala> b += "there"
val res1: b.type = ArrayBuffer(hi, there)

scala> val hiThere = b.toArray
val hiThere: Array[String] = Array(hi, there)

scala> hi
val res2: Array[String] = Array(hi)
```

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/UserManagementServiceImpl.scala [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      request: RevokeUserRightsRequest
+  ): Future[RevokeUserRightsResponse] = record(request)(RevokeUserRightsResponse.defaultInstance)
+
+  override def listUserRights(request: ListUserRightsRequest): Future[ListUserRightsResponse] =
+    record(request)(ListUserRightsResponse.defaultInstance)
+}
+
+object UserManagementServiceImpl {
+
+  def createWithRef(
```
Comment (S11001001) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/UserManagementServiceImpl.scala:56]:
```suggestion
  // for testing only
  private[helpers] def createWithRef(
```

**Review by S11001001 — APPROVED** [review]
👍

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/LedgerServices.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/LedgerServices.scala:240]: 😢

**Review by meiersi-da — APPROVED** [review]
Thanks a lot :pray: LGTM. What a brutal amount of boilerplate :cry:

**Comment by gohchanb** [comment]
I've noticed that a GetUserResponse class is missing from the java bindings

**Comment by stefanobaghino-da** [comment]
> I've noticed that a GetUserResponse class is missing from the java bindings

@gohchanb Thanks for raising this. While this is not strictly a problem (as the `GetUserResponse` only returns the `User`, so no information is lost), we wanted to explicitly wrap types everywhere to make sure to be future-proof in case we needed to add fields to the response. At this time, this change would require users to adapt their code, so I'll need to ask whether we want to act on this or not. We are probably still in a good place to make this change, given that user management has been introduced in 2.0.

**Pull request merged**